dexlib2/src/main/java/com/android/tools/smali/dexlib2/writer/builder/BuilderAnnotationSetPool.java

```diff
@@ -65,12 +64,9 @@ public BuilderAnnotationSetPool(@Nonnull DexBuilder dexBuilder) {
         }
 
         BuilderAnnotationSet annotationSet = new BuilderAnnotationSet(
-                ImmutableSet.copyOf(Iterators.transform(annotations.iterator(),
-                        new Function<Annotation, BuilderAnnotation>() {
-                            @Nullable @Override public BuilderAnnotation apply(Annotation input) {
-                                return dexBuilder.annotationSection.internAnnotation(input);
-                            }
-                        })));
+            Collections.unmodifiableSet(annotations.stream()
```

Comment:
Don't remember which Android versions you were targeting with smali, but using streams requires Android 7 (API level 24) or library desugaring (https://developer.android.com/studio/write/java8-support#library-desugaring).

Reply:
This refactor spans only dexlib2, and the added device target has min_sdk_version 33. We should be in the clear. But if you know of any possible usages that might want to target further back, we can revisit.

Reply:
SGTM